### ISSUE TITLE
Ability to specify a custom plugin folder

### ISSUE THREAD

**Issue by iloire** [itteco/iframely#143]
Story: "As a developer, I would like to be able to use a custom plugin folder, so I can override the behaviour of certain plugins".

It could be as simple as:

```
// iframely/config.local.js

  CUSTOM_PLUGINS: __dirname + '/custom-plugins',
```

```
//iframely/lib/loader/pluginLoader.js line 190

             pluginDeclaration.id = getFileName(bits[bits.length - 1]).toLowerCase();
             if (pluginDeclaration.id in plugins) {
-                console.error("Duplicate plugin id (filename)", pluginPath);
-                return;
+
+                if (CONFIG.CUSTOM_PLUGINS && bits.indexOf(CONFIG.CUSTOM_PLUGINS) > -1) {
+                  console.error("Plugin '" + pluginDeclaration.id + "' overridden by", pluginPath);
+                  delete plugins[pluginDeclaration.id]
+                } else {
+                  console.error("Duplicate plugin id (filename)", pluginPath);
+                  return;
+                }
             }
```

```
// iframely/lib/loader/pluginLoader.js line 431

                 loadPlugins('plugins', 'domains');
                 loadPlugins('plugins', 'custom');
                 loadPlugins('plugins', 'links');
                 loadPlugins('plugins', 'meta');
                 loadPlugins('plugins', 'templates');

+                if (CONFIG.CUSTOM_PLUGINS) {
+                  if (fs.existsSync(CONFIG.CUSTOM_PLUGINS)) {
+                    loadPlugins(CONFIG.CUSTOM_PLUGINS, 'domains');
+                    loadPlugins(CONFIG.CUSTOM_PLUGINS, 'custom');
+                    loadPlugins(CONFIG.CUSTOM_PLUGINS, 'links');
+                    loadPlugins(CONFIG.CUSTOM_PLUGINS, 'meta');
+                    loadPlugins(CONFIG.CUSTOM_PLUGINS, 'templates');
+                  } else {
+                    console.warn('Custom plugin folder "' + CONFIG.CUSTOM_PLUGINS + '" not found.');
+                  }
+                }
```

I can work on a proper PR with tests if this is of interest.

**Comment by iparamonau** [comment]
Thanks, this story is long overdue. We planned to do it exactly the way you described. We're will be publishing v1.0.0 on Friday and I'd like this one to be included. If you have time to do it before Friday - pls submit the PR. Otherwise we'll just add it ourselves. Cheers!

**Comment by iloire** [comment]
A PR will be raise in the next hour...

**Comment by iparamonau** [comment]
Resolved via #145. Thanks!
